src/Components/Patient/PatientRegister.tsx

```diff
@@ -228,6 +229,19 @@ export const PatientRegister = (props: PatientRegisterProps) => {
   const headerText = !id ? "Add Details of Patient" : "Update Patient Details";
   const buttonText = !id ? "Add Patient" : "Save Details";
 
+  const fetchAllVaccines = useCallback(async () => {
+    setIsVaccineLoading(true);
+    const { res, data } = await request(routes.getVaccines);
+    if (res?.ok && data?.results) {
+      setVaccinesList(data.results);
+    }
+    setIsVaccineLoading(false);
+    console.log(data?.results);
+    return data?.results || [];
+  }, []);
+  useEffect(() => {
+    fetchAllVaccines();
+  }, []);
```

Comment:
use useQuery which keeps the code clean and no need of extra states like vaccinesList and isVaccineLoading